### DISCUSSION TITLE
Enhancements in Play and pause button: Reducing UI redundancy by refactoring the Run and stop buttons into one.

### DISCUSSION THREAD

**Discussion by grvup** [processing/p5.js-web-editor#2789] [General]
### Increasing Access

This enhancement aims to simplify user interactions

### Feature enhancement details

Currently there is two different button for 'play' and 'pause'. This feature will introduce a single button that dynamically changes its label between 'Play' and 'Pause' based on the current state.
![p5](https://github.com/processing/p5.js-web-editor/assets/128596203/1655b330-2aba-4816-a58b-e33174022130)

**Comment by welcome** [comment]
Welcome! 👋 Thanks for opening your first issue here! And to ensure the community is able to respond to your issue, be sure to follow the issue template if you haven't already.

**Comment by HrxSrv** [comment]
The enhancement sounds legit, but having those two separate buttons actually solves a problem that arises due to a single button. The button will be responsive instantaneously; however, the task it will perform is inherently asynchronous in nature, which raises the issue of delay in touch and response time of the button.

**Comment by SableRaf** [comment]
Thanks for your suggestion!

The two-button design, with separate 'Play' and 'Pause' buttons, is a deliberate choice carried over from Processing. It makes the interface immediately readable and consistent regardless of state. It is an established convention and changing it now might confuse long-time users.

Besides, combined buttons have a number of issues such as potential confusion on whether the button displays its current state or the action it will perform next, and the risk of accidental double-clicks, causing the play/pause button to not respond as expected.

I found some interesting reflections on this topic here and some of the points above are borrowed from that thread: https://ux.stackexchange.com/questions/1215/separation-of-play-and-pause-buttons-for-playing-media
